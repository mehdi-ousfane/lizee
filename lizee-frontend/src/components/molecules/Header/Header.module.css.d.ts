declare const styles: {
  readonly "headerContainer": string;
  readonly "lizeeIconContainer": string;
  readonly "navigationIcons": string;
  readonly "userIconContainer": string;
  readonly "helpIconContainer": string;
};
export = styles;

