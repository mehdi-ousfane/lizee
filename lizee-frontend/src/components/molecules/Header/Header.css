.headerContainer {
  display: flex;
}

.lizeeIconContainer {
  justify-content: flex-start;
  cursor: pointer;
}

.navigationIcons {
  justify-content: flex-end;
}

.userIconContainer,
.helpIconContainer {
  cursor: pointer;
}
