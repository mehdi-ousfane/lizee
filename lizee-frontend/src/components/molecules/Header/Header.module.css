.headerContainer {
  display: flex;
  height: 2.8rem;
}

.lizeeIconContainer {
  margin-left: 6rem;
  flex: 1;
}

.lizeeIconContainer > svg {
  cursor: pointer;
}

.navigationIcons {
  display: flex;
  margin-right: 7rem;
  flex: 1;
  justify-content: flex-end;
  align-items: center;
}

.userIconContainer,
.helpIconContainer {
  display: flex;
  align-items: center;
}

.userIconContainer > p,
.helpIconContainer > p {
  margin: 0 0.3rem;
  font-family: 'Patron webfont', sans-serif;
  font-size: 0.8rem;
}

.userIconContainer > p,
svg,
.helpIconContainer > p,
svg {
  cursor: pointer;
}
