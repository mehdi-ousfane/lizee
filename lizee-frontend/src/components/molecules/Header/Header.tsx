import HelpIcon from '../../icons/HelpIcon/HelpIcon';
import LizeeIcon from '../../icons/LizeeIcon/LizeeIcon';
import UserIcon from '../../icons/UserIcon/UserIcon';
import styles from './Header.css';

const Header = () => {
  return (
    <div className={styles.headerContainer}>
      <div className={styles.lizeeIconContainer}>
        <LizeeIcon />
      </div>
      <div className={styles.navigationIcons}>
        <div className={styles.userIconContainer}>
          <UserIcon />
        </div>
        <div className={styles.helpIconContainer}>
          <HelpIcon />
        </div>
      </div>
    </div>
  );
};

export default Header;
