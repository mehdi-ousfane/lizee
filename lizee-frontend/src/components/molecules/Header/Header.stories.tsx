import { Meta } from '@storybook/react/types-6-0';
import Header from './Header';

export default {
  title: 'molecules/Header',
  component: Header,
} as Meta;

export const Primary = () => <Header />;
