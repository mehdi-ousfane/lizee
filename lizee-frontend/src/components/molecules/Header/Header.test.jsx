import { render } from '@testing-library/react';
import Header from './Header';

describe('<Header />', () => {
  test('should render without crashing', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
