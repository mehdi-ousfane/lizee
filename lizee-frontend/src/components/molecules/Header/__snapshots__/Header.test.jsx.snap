// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render without crashing 1`] = `
<div>
  <header
    class="headerContainer"
  >
    <div
      class="lizeeIconContainer"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 634 288"
        width="83"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.778 167.596H301.577V191H232.702V73.9793H257.778V167.596ZM325.013 55.5903C329.693 55.5903 333.371 56.9834 336.046 59.7697C338.721 62.4444 340.058 66.0665 340.058 70.6359C340.058 75.0938 338.721 78.7159 336.046 81.5021C333.371 84.2883 329.693 85.6814 325.013 85.6814C320.22 85.6814 316.431 84.344 313.645 81.6692C310.97 78.883 309.633 75.2052 309.633 70.6359C309.633 66.0665 310.97 62.4444 313.645 59.7697C316.32 56.9834 320.109 55.5903 325.013 55.5903ZM336.715 191H313.311V97.3835H336.715V191ZM430.794 113.599L386.827 169.268H430.794V191H355.064V174.784L398.864 118.782H355.064V97.3835H430.794V113.599ZM453.396 109.086C461.754 99.9468 472.286 95.3774 484.991 95.3774C497.696 95.3774 508.284 99.9468 516.754 109.086C525.224 118.113 529.459 129.871 529.459 144.359L529.292 151.046H465.265C465.265 156.841 467.327 161.689 471.45 165.59C475.574 169.379 480.589 171.274 486.496 171.274C491.845 171.274 496.916 169.825 501.709 166.927C506.501 163.918 510.011 160.519 512.24 156.73H513.411L524.11 176.79C521.658 180.468 517.033 184.09 510.234 187.657C503.547 191.223 495.635 193.006 486.496 193.006C472.899 193.006 461.866 188.66 453.396 179.967C445.037 171.162 440.858 159.46 440.858 144.86C440.858 130.149 445.037 118.224 453.396 109.086ZM498.532 122.459C495.077 118.893 490.564 117.11 484.991 117.11C479.419 117.11 475.017 118.837 471.785 122.292C468.553 125.636 466.491 129.313 465.599 133.326H504.383C503.938 129.536 501.987 125.914 498.532 122.459ZM552.001 109.086C560.36 99.9468 570.892 95.3774 583.597 95.3774C596.302 95.3774 606.89 99.9468 615.36 109.086C623.83 118.113 628.065 129.871 628.065 144.359L627.898 151.046H563.871C563.871 156.841 565.932 161.689 570.056 165.59C574.18 169.379 579.195 171.274 585.101 171.274C590.451 171.274 595.522 169.825 600.314 166.927C605.106 163.918 608.617 160.519 610.846 156.73H612.016L622.715 176.79C620.263 180.468 615.638 184.09 608.84 187.657C602.153 191.223 594.24 193.006 585.101 193.006C571.505 193.006 560.471 188.66 552.001 179.967C543.643 171.162 539.463 159.46 539.463 144.86C539.463 130.149 543.643 118.224 552.001 109.086ZM597.138 122.459C593.683 118.893 589.169 117.11 583.597 117.11C578.025 117.11 573.622 118.837 570.39 122.292C567.158 125.636 565.097 129.313 564.205 133.326H602.989C602.543 129.536 600.593 125.914 597.138 122.459Z"
          fill="#2933E7"
        />
        <path
          clip-rule="evenodd"
          d="M173.002 191.702H101.73H96.0472H62.4902H56.1481H48.0236C40.0466 191.702 33.557 185.212 33.557 177.235V48.0212C33.557 40.0442 40.0466 33.557 48.0236 33.557C56.0006 33.557 62.4902 40.0442 62.4902 48.0212V183.32H96.0472V48.0212C96.0472 21.5416 74.5032 0 48.0236 0C21.544 0 0 21.5416 0 48.0212V177.235C0 203.715 21.544 225.259 48.0236 225.259H56.1481H62.4902H96.0472H101.73H173.002C180.979 225.259 187.468 231.746 187.468 239.725C187.468 247.7 180.979 254.192 173.002 254.192H110.511C102.537 254.192 96.0472 247.7 96.0472 239.725V233.64H62.4902V239.725C62.4902 266.205 84.0318 287.747 110.511 287.747H173.002C199.481 287.747 221.025 266.205 221.025 239.725C221.025 213.246 199.481 191.702 173.002 191.702Z"
          fill="url(#paint0_linear)"
          fill-rule="evenodd"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear"
            x1="0"
            x2="251.117"
            y1="0"
            y2="251.117"
          >
            <stop
              stop-color="#2733E8"
            />
            <stop
              offset="1"
              stop-color="#ED4D7B"
            />
          </lineargradient>
        </defs>
      </svg>
    </div>
    <div
      class="navigationIcons"
    >
      <div
        class="userIconContainer"
      >
        <svg
          height="28"
          viewBox="0 0 512 512"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 0C114.6 0 0 114.6 0 256s114.6 256 256 256s256-114.6 256-256S397.4 0 256 0zM256 128c39.77 0 72 32.24 72 72S295.8 272 256 272c-39.76 0-72-32.24-72-72S216.2 128 256 128zM256 448c-52.93 0-100.9-21.53-135.7-56.29C136.5 349.9 176.5 320 224 320h64c47.54 0 87.54 29.88 103.7 71.71C356.9 426.5 308.9 448 256 448z"
          />
        </svg>
        <p>
          My account
        </p>
      </div>
      <div
        class="helpIconContainer"
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 28 28"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.2516 6.87651C14.0444 6.75402 14.8568 6.84761 15.5999 7.14788C16.3432 7.44821 16.9902 7.94457 17.4673 8.58519C17.9446 9.22605 18.2327 9.98569 18.2968 10.7804C18.3609 11.5751 18.1983 12.3706 17.8286 13.0782C17.4592 13.7856 16.8982 14.3762 16.2107 14.7872C16.0284 14.8961 15.8303 14.9983 15.6708 15.0806C15.625 15.1043 15.5824 15.1263 15.5442 15.1463C15.349 15.2489 15.2222 15.3239 15.1283 15.3977C15.0435 15.4642 15.007 15.5137 14.986 15.5549C14.9658 15.5948 14.9331 15.68 14.9331 15.856V17.1095C14.9331 17.4409 14.6644 17.7095 14.3331 17.7095H13.5331C13.2017 17.7095 12.9331 17.4409 12.9331 17.1095V15.856C12.9331 15.4142 13.0194 15.0103 13.203 14.6489C13.386 14.2887 13.6383 14.0247 13.8932 13.8246C14.1388 13.6317 14.4023 13.487 14.6141 13.3758C14.6794 13.3415 14.7392 13.3106 14.795 13.2818C14.9406 13.2066 15.0606 13.1445 15.1844 13.0705C15.5592 12.8465 15.8598 12.5278 16.0559 12.1523C16.2519 11.777 16.3369 11.3581 16.3033 10.9412C16.2696 10.5242 16.1183 10.1222 15.8633 9.77982C15.6081 9.43718 15.2584 9.16698 14.8506 9.00221C14.4427 8.83738 13.9947 8.78543 13.557 8.85306C13.1193 8.92069 12.7119 9.10473 12.3791 9.38198C12.2046 9.52735 12.0544 9.69522 11.9317 9.87962C11.7482 10.1555 11.4141 10.3322 11.0997 10.2275L10.3407 9.97467C10.0263 9.86996 9.8527 9.5276 10.0006 9.23107C10.2652 8.7007 10.6389 8.22871 11.0989 7.84542C11.7135 7.33335 12.4587 6.99902 13.2516 6.87651Z"
            fill="black"
          />
          <path
            d="M13.1111 19.8763C13.1111 19.6002 13.335 19.3763 13.6111 19.3763H14.2587C14.5348 19.3763 14.7587 19.6002 14.7587 19.8763V20.5239C14.7587 20.8 14.5348 21.0239 14.2587 21.0239H13.6111C13.335 21.0239 13.1111 20.8 13.1111 20.5239V19.8763Z"
            fill="black"
          />
          <path
            clip-rule="evenodd"
            d="M14 1C6.8203 1 1 6.8203 1 14C1 21.1797 6.8203 27 14 27C21.1797 27 27 21.1797 27 14C27 6.8203 21.1797 1 14 1ZM3 14C3 7.92487 7.92487 3 14 3C20.0751 3 25 7.92487 25 14C25 20.0751 20.0751 25 14 25C7.92487 25 3 20.0751 3 14Z"
            fill="black"
            fill-rule="evenodd"
          />
        </svg>
        <p>
          Help
        </p>
      </div>
    </div>
  </header>
</div>
`;
